### PR TITLE
fix RandAffine with randomize==False

### PR DESCRIPTION
**Pull request by rijobro**
Fixes https://github.com/Project-MONAI/MONAI/issues/4303.

### Description
Pass the `randomize` flag on to `RandAffineGrid` so that it randomises at the correct moment. Unit test added to ensure this.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.

### CODE REVIEW

**Comment by rijobro** [comment]
@fluentgcc Thanks for spotting the bug, this should solve the issue.

File: monai/transforms/spatial/array.py [hunk 1]
```diff
@@ -1530,7 +1533,8 @@ def __call__(
         Returns:
```
Comment (Nic-Ma) [inline monai/transforms/spatial/array.py:1533]: Could you please help add the new arg to doc-string?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick fix.
Looks good to me, put minor comments inline.

Thanks.

File: monai/transforms/spatial/array.py (hunk 1 above)
Comment (rijobro) [reply monai/transforms/spatial/array.py:1535]: Good point, done!

**Pull request merged**